### DISCUSSION TITLE
Change the SSO title

### DISCUSSION THREAD

**Discussion by rougeo** [directus/directus#14111] [Archive]
In order to use an auth server for open id like keycloak, we need to use the Github auth provider with the keycloal url. This part is working fine. But on the login page we still have the button "Log in with Github". Would it be possible to change this text to a custom one. So far I am try to replace the css "sso-title" with the text I want.

**Comment by azrikahar** [comment]
> we need to use the Github auth provider with the keycloal url.

The current SSO integrations are agnostic, so there isn't technically GitHub auth provider (unless I'm misunderstanding the description here).

Assuming what you meant is you were configuring it such as:

```env
AUTH_PROVIDERS="github"

AUTH_GITHUB_DRIVER="..."
AUTH_GITHUB_CLIENT_ID="..."
AUTH_GITHUB_CLIENT_SECRET="..."
# other relevant configurations
```

then you should do the following instead:

```env
AUTH_PROVIDERS="keycloak"

AUTH_KEYCLOAK_DRIVER="..."
AUTH_KEYCLOAK_CLIENT_ID="..."
AUTH_KEYCLOAK_CLIENT_SECRET="..."
# other relevant configurations
```

So that it'll show up as `Log in with Keycloak` 👍 There's also a useful list of example configs over here, with this link bringing you to a `keycloak` example: https://docs.directus.io/self-hosted/sso-examples/#keycloak

**Reply by rougeo** [reply]
Hello thanks, I didn't saw that KEYCLOAK config exemple, I just used the Github one with the keycloak parameters and it just worked. I will fix that on my side. So back to the initial question, is there a way to replace the text "log in with keycloak " ?

**Reply by aidenfoxx** [reply]
As @azrikahar said, you only need to change the provider name in your config file and the text will update in the app.

So changing your config to `AUTH_PROVIDERS="my_provider_name"` and the associated configs e.g. `AUTH_MY_PROVIDER_NAME_DRIVER="..."` will display as "Log in with my provider name" in the app.

**Reply by joselcvarela** [reply]
Just to let you know that there's now a pending PR to allow a custom label on App for a provider:
https://github.com/directus/directus/pull/15673
